tests/next_tests/iterator_tests/test_trace_shifts.py

```diff
@@ -120,15 +121,47 @@ def test_neighbors():
 
 def test_reduce():
     testee = ir.StencilClosure(
-        stencil=ir.FunCall(
-            fun=ir.SymRef(id="reduce"),
-            args=[ir.SymRef(id="plus"), ir.SymRef(id="init")],
-        ),
+        # λ(inp) → reduce(plus, init)(·inp)
+        stencil=ir.Lambda(params=[ir.Sym(id="inp")], expr=ir.FunCall(
+            fun=ir.FunCall(fun=ir.SymRef(id="reduce"),
+                           args=[ir.SymRef(id="plus"), ir.SymRef(id="init")]),
+            args=[ir.FunCall(fun=ir.SymRef(id="deref"), args=[ir.SymRef(id="inp")])])),
+        inputs=[ir.SymRef(id="inp")],
+        output=ir.SymRef(id="out"),
+        domain=ir.FunCall(fun=ir.SymRef(id="cartesian_domain"), args=[]),
+    )
+    expected = {"inp": [()]}
+
+    actual = dict()
+    TraceShifts().visit(testee, shifts=actual)
+    assert actual == expected
+
+
+def test_shifted_literal():
+    "Test shifting an applied lift of a stencil returning a constant / literal works."
+    testee = ir.StencilClosure(
+        # λ(x) → ·⟪Iₒ, 1ₒ⟫((↑(λ() → 1))())
+        stencil=im.lambda__("x")(im.deref_(im.shift_("I", 1)(im.lift_(im.lambda__()(1))()))),
         inputs=[ir.SymRef(id="inp")],
         output=ir.SymRef(id="out"),
         domain=ir.FunCall(fun=ir.SymRef(id="cartesian_domain"), args=[]),
     )
-    expected = dict()
```

Comment:
Note to the reviewer: This was actually wrong before. The stencil does a deref at the center point.